nix/tools/withTools.nix

```diff
@@ -426,6 +427,10 @@ let
           ${monitorPid} "$pid" > "$_arg_monitor" &
         fi
 
+        if [[ -n "$_arg_sleep" ]]; then
+          sleep "$_arg_sleep"
+        fi
```

Comment:
Looks like this argument has a default of 0, so the condition will always be true. Should work to just call `sleep` all the time - with 0 as argument if the default is used.